sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mlme.c

```diff
@@ -299,19 +280,24 @@ void gnrc_lorawan_process_fopts(gnrc_lorawan_t *mac, uint8_t *fopts, size_t size
         return;
     }
 
-    lorawan_buffer_t buf;
-    _buffer_reset(&buf, fopts, size);
+    uint8_t ret = 0;
+    int (*cb)(gnrc_lorawan_t*, uint8_t *p) = NULL;
 
-    while (buf.index < buf.size) {
-        switch (*(buf.data)) {
+    for(uint8_t pos = 0; pos < size; pos += ret) {
+        switch (fopts[pos]) {
             case GNRC_LORAWAN_CID_LINK_CHECK_REQ_ANS:
-                if (_mlme_link_check_ans(mac, &buf) < 0) {
-                    return;
-                }
+                ret += 3;
+                cb = _mlme_link_check_ans;
```

Comment:
this is hard to get, even though I'm looking at this in the morning, it took me a moment to realise that this just sets a function pointer which is then called later. 

I would suggest to simply call that function here, and not doing this redirection via a function pointer - this is hard to read.

Reply:
as described before, I have to check it doesn't overflow before calling the function :(. I would need to add checks for each function in that case